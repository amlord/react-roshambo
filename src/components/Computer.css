.Computer {
    vertical-align: top;
    font-size: 1rem;
    display: inline-block;
    width: 31.578947368421053%;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
}

.Computer__title {
    font-size: 1em;
    margin: 0;
    padding: 5px;
    background: #ececed;
}

.Computer__choice {
    height: 238px;
    margin: 10px;
    border-radius: 4px;
    background: #f2f2f2;

    display: flex;
    align-items: center;
    justify-content: center;
}

.Computer__ChosenShape {
    margin: 0;
}