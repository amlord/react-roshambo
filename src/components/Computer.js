import React, { Component } from 'react';

import './Computer.css';
import { Translate } from '../helpers/Constants';

class Computer extends Component {
  render() {
    return (
      <section className="Computer">
        <header>
          <header>
            <h1>Computer</h1>
          </header>
          <div className="chosenShape">
            <h2>{Translate.choice(this.props.chosen)}</h2>
          </div>
        </header>
      </section>
    );
  }
}

export default Computer;
