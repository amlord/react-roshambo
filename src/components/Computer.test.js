import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Computer from './Computer';

describe('Computer', () => {
  const computer = shallow(<Computer />);

  it('renders correctly', () => {
    expect(computer).toMatchSnapshot();
  });
});