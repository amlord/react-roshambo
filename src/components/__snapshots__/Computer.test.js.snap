// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Computer renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Computer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <header>
        <header>
                <h1
                        className="Computer__title"
                >
                        Computer
                </h1>
        </header>
        <div
                className="Computer__choice"
        >
                <h2
                        className="Computer__ChosenShape"
                />
        </div>
</header>,
      "className": "Computer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
                        className="Computer__title"
            >
                        Computer
            </h1>
</header>,
          <div
            className="Computer__choice"
>
            <h2
                        className="Computer__ChosenShape"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="Computer__title"
>
              Computer
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Computer",
              "className": "Computer__title",
            },
            "ref": null,
            "rendered": "Computer",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="Computer__ChosenShape"
/>,
            "className": "Computer__choice",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "Computer__ChosenShape",
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <header>
          <header>
                    <h1
                              className="Computer__title"
                    >
                              Computer
                    </h1>
          </header>
          <div
                    className="Computer__choice"
          >
                    <h2
                              className="Computer__ChosenShape"
                    />
          </div>
</header>,
        "className": "Computer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header>
              <h1
                            className="Computer__title"
              >
                            Computer
              </h1>
</header>,
            <div
              className="Computer__choice"
>
              <h2
                            className="Computer__ChosenShape"
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="Computer__title"
>
                Computer
</h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Computer",
                "className": "Computer__title",
              },
              "ref": null,
              "rendered": "Computer",
              "type": "h1",
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2
                className="Computer__ChosenShape"
/>,
              "className": "Computer__choice",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "Computer__ChosenShape",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            "type": "div",
          },
        ],
        "type": "header",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
